### DISCUSSION TITLE
Does RHF only validate fields when they are dirty?

### DISCUSSION THREAD

**Discussion by valerio-pescatori** [react-hook-form/react-hook-form#11223] [Q&A]
I have a `useFieldArray` of objects, every input of the object has a validation rule defined.
I also have an 'add item' button and I want it to be disabled if the array is not valid. I'm using `getFieldState` to check the validity of the array (yes, i'm correctly subscribing to `errors` in the `formState`), but the validation won't be triggered until I touch one of the inputs in the array, so the button appears enabled even if the form is not valid.

I tried combining the `invalid` and `isDirty` properties from `getFieldState` to disable the 'add item' button but this causes more problems (e.g if I pre-populate the form with values from a fetch request, it might be valid even if it is not dirty, so the button will be disabled while it shouldn't).

I also tried calling `trigger` to validate the fields when mounting the component or when `append`ing an item to the array but it still won't change the invalid prop from getFieldState.   Any ideas on how to solve this? thanks

**Comment by ktalalykin** [comment]
it is hard to say what is happening without code/sandbox provided. It depends on the sort of validation you have, how you set up the form etc. If you can provide sandbox, that can help!

**Reply by valerio-pescatori** [reply]
It's a very big and complex form, I'll try to reduce it as much as I can and post the sandbox asap! ty
